### PR TITLE
`compute_intersection` with CGAL

### PR DESCRIPTION
**Pull request by fdrmrc**
In this PR I've added CGAL and the `compute_intersection(cell0,cell1,degree,mapping0,mapping1)` free function. Some simple tests in co-dimension one have been added.  Some more non-trivial tests in co-dimension 0 are in progress.

I think we could add the actual test with Fictitious Domain method in a separate PR, once this one is okay.

### CODE REVIEW

**Comment by fdrmrc** [comment]
Co-dimension 0 tests have been added.

**Review by luca-heltai — CHANGES_REQUESTED** [review]
Can you rebase on top of the new master? I have forced push on main to remove from the repository the lecture notes. I've left some comments. As soon as you split header and source, and leave cgal include out of the header, we can merge, and make corrections later.

File: include/compute_intersection_of_cells.h [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+// distribution.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef compute_intersction_of_cells_h
+#define compute_intersction_of_cells_h
+
+
+// CGAL header files and typedefs
```
[the remark this answers is not in the document]
Comment (fdrmrc) [reply include/compute_intersection_of_cells.h:21]: Thanks, so this .h file will contain only the `#include "compute_linear_transformation.h"`and the header guards, as all functions are using CGAL

File: include/compute_intersection_of_cells.h [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Polygon_2.h>
+#include <CGAL/Polygon_with_holes_2.h>
+#include <CGAL/Triangle_2.h>
+#include <CGAL/Triangulation_2.h>
+
+typedef CGAL::Exact_predicates_exact_constructions_kernel_with_sqrt Kernel;
+typedef CGAL::Polygon_2<Kernel>            CGAL_Polygon;
+typedef CGAL::Polygon_with_holes_2<Kernel> Polygon_with_holes_2;
+typedef CGAL_Polygon::Point_2              CGAL_Point;
```
[the remark this answers is not in the document]
Comment (fdrmrc) [reply include/compute_intersection_of_cells.h]: Yes

**Pull request merged**